sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathClient.java

```diff
@@ -113,7 +190,10 @@ String getObjectPath() {
      * @return The name of the object.
      */
     String getObjectName() {
-        return dataLakePathAsyncClient.getObjectName();
+        // Split on / in the path
+        String[] pathParts = getObjectPath().split("/");
```

Comment:
Not related to this change as this logic is being copied.

Couldn't this be computed ahead of time instead of each time this API is called?